File: 1413. Minimum Value to Get Positive Step by Step Sum/solution.cpp
class Solution {
public:
  int minStartValue(vector<int>& nums) {
    int start = 1;
    int total = start;
    for (int index = 0; index < nums.size(); index++){
      int num = nums[index];
      total += num;
      if (total < 1){
        int missing = 1 - total;
        start += missing;
        total += missing;
      }
    }

    return start;
  }
};

File: 1413. Minimum Value to Get Positive Step by Step Sum/solution.py
class Solution(object):
  def minStartValue(self, nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    start = 1
    total = start
    for num in nums:
      total += num
      if total < 1:
        missing = 1 - total
        start += missing
        total += missing
    return start